lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -580,12 +583,153 @@ def process(self, temperature, relative_humidity, pressure):
+                before returning wet bulb temperatures.
+            falling_level_threshold (float):
+                The cutoff threshold for the Wet-bulb integral used
+                to calculate the falling snow level.We are integrating to the
+                threshold that is presumed to indicate
+                the level at which snow has melted back to rain. Above
+                this level we should have falling snow.
+
+        """
+        self.precision = precision
+        self.wet_bulb_integral_plugin = (
+            WetBulbTemperatureIntegral(precision=precision))
+        self.falling_level_threshold = falling_level_threshold
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<FallingSnowLevel: precision:'
+                  '{}, falling_level_threshold:{}>'.format(
+                      self.precision,
+                      self.falling_level_threshold))
+        return result
+
+    def find_falling_level(self, wb_int_data, orog_data, height_points):
+        """
+        Find the falling snow level by finding the level of the wet-bulb
+        integral data at the required threshold.
+
+        Args:
+            wb_int_data (np.array):
+                Wet bulb integral data on heights
+            orog_data (np.array):
+                Orographic data
+            heights (np.array):
+                heights agl
+
+        Returns:
+            snow_level_data (np.array):
+                Falling snow level data asl.
+
+        """
+        # Create cube of heights above sea level for each height in
+        # the wet bulb integral cube.
+        asl = wb_int_data.copy()
+        for i, height in enumerate(height_points):
+            asl[i, ::] = orog_data + height
+
+        # Calculate falling snow level above sea level by
+        # finding the level corresponding to the falling_level_threshold.
+        # Interpolate returns an array with height indice
+        #  for falling_level_threshold so we take the 0 index
+        snow_level_data = interpolate(np.array([self.falling_level_threshold]),
+                                      wb_int_data, asl, axis=0)[0]
+        return snow_level_data
+
+    @staticmethod
+    def fill_in_missing_data(snow_level_data):
+        """
+        Fill in missing data
+
+        Args:
+            snow_level_data (np.array):
+                Falling snow level data.
+
+        Returns:
+            snow_level_updated (np.array):
+                Falling snow level data with missing data added.
+
+        """
+        points = np.where(np.isfinite(snow_level_data))
+        values = snow_level_data[points]
+        ynum, xnum = snow_level_data.shape
+        (y_points, x_points) = np.mgrid[0:ynum, 0:xnum]
+        snow_level_updated = griddata(points, values, (y_points, x_points),
+                                      method='linear', fill_value=0.0)
```

Comment:
I think we need the line above to continue to return NaN values. Then extra code is required to replace these NaN values with either "start_point + orog" if highest level of the wet bulb integral is greater than 90 Kelvin-metres (meaning that we started looking below the freezing level) or "end_point + orog" everywhere else. (Don't check for negative values, there won't be any).